alternator/conditions.cc

```diff
@@ -143,9 +143,44 @@ static void verify_operand_count(const rjson::value* array, const size_check& ex
     }
 }
 
+// It's not enough to compare underlying JSON objects when comparing sets,
+// as internally they're stored in an array, and the order of elements is
+// not important in set equality. See issue #5021
+static bool check_EQ_for_sets(const rjson::value& s1, const rjson::value& s2) {
+    const rjson::value* set1 = unwrap_set(s1).second;
+    const rjson::value* set2 = unwrap_set(s2).second;
+
+    if (!set1 || !set2) {
+        return !set1 && !set2;
+    }
+    if (set1->Size() != set2->Size()) {
+        return false;
+    }
+    std::set<rjson::value, rjson::single_value_comp> set1_raw;
+    for (auto it = set1->Begin(); it != set1->End(); ++it) {
```

Comment:
Nit: isn't there a `set` constructor that takes a pair of iterators?

Reply:
there is, but copying an rjson value must be explicit, so it would fail. While we're at it, I didn't really like the fact that I need to actually copy the rjson value just to add it to a set, but I decide to ignore it as it's probably not a super hot path. I changed my mind though, I'll send v2 which uses a set of pointers, so that copying is elided.

Reply:
done